flink-core/src/main/java/org/apache/flink/metrics/MetricRegistry.java

```diff
@@ -64,6 +66,15 @@ public MetricRegistry(Configuration config) {
 		}
 		this.scopeFormats = scopeFormats;
 
+		char delim;
+		try {
+			delim = config.getString(ConfigConstants.METRICS_SCOPE_DELIMITER, ".").charAt(0);
```

Comment:
Why do we limit the delimiter to being a character. Couldn't it be useful for the user to use a string as a delimiter?